src/app/main/component/shared/components/form-base/form-base.component.ts

```diff
@@ -64,4 +65,23 @@ export class FormBaseComponent implements ComponentCanDeactivate {
       }
     }
   }
+
+  cancelUBS(): void {
+    if (this.getFormValues()) {
+      const matDialogRef = this.dialog.open(WarningPopUpComponent, this.popupConfig);
+
+      matDialogRef
+        .afterClosed()
+        .pipe(take(1))
+        .subscribe((confirm) => {
+          if (confirm) {
+            this.areChangesSaved = true;
+            this.router.navigate([this.previousPath]);
+          }
```

Comment:
if you need only truthy value - you can filter it with help of rxjs -> https://stackoverflow.com/questions/53953015/using-rxjs-with-filterboolean-for-queries